lib/shopify-cli/commands/help.rb

```diff
@@ -21,27 +21,46 @@ def call(args, _name)
         end
 
         preamble = <<~MESSAGE
-          CLI to help build Shopify apps faster.
-
           Use {{command:#{ShopifyCli::TOOL_NAME} help <command>}} to display detailed information about a specific command.
 
-          {{bold:Available commands}}
+          {{bold:Available core commands:}}
+
         MESSAGE
         @ctx.puts(preamble)
 
-        visible_commands = ShopifyCli::Commands::Registry
-          .resolved_commands
-          .select { |_name, c| !c.hidden }
-          .sort
+        core_commands.each do |name, klass|
+          next if name == 'help'
+          @ctx.puts("{{command:#{name}}}: #{klass.help}\n")
+        end
+
+        return unless inside_supported_project?
 
-        visible_commands.each do |name, klass|
+        @ctx.puts("{{bold:Project: #{File.basename(Dir.pwd)} (#{project_type_name})}}")
+        @ctx.puts("{{bold:Available commands for #{project_type_name} projects:}}\n\n")
+
```

Comment:
`#{project_type_name}` is shown on two consecutive lines, which seems a little redundant.  Maybe something like:
```
@ctx.puts("{{bold:Available commands for #{project_type_name} project}} {{cyan:#{File.basename(Dir.pwd)}:}}\n\n")
```
or
```
@ctx.puts("{{bold:Available commands for}} {{cyan:#{File.basename(Dir.pwd)} (#{project_type_name}):\n\n")
```

Also (perhaps make an issue for this for another PR), it would be nicer to have something like `Project.project_name` (for example) instead of `File.basename(Dir.pwd)` to convey the project name.

Reply:
Any thoughts on this @attaboy?

> Also (perhaps make an issue for this for another PR), it would be nicer to have something like Project.project_name (for example) instead of File.basename(Dir.pwd) to convey the project name.

Yup, I agree, I'll handle this in a follow-up.

Reply:
It is a little redundant agree, but I think it's useful redundancy — we're giving the user some crucial information here, and helping reinforce some concepts that I don't think are terribly obvious — that there is a concept of a "project", and there are different types of projects, and that different types of projects have specific commands.

I considered `Available commands for this project type:` for the second line, but IMO it's just slightly less clear. (I don't feel that strongly about it though.)

Reply:
Fair enough - if, from a UX perspective, it's useful to keep it to convey a message/concept, I'm good.